homeassistant/components/mysensors/config_flow.py

```diff
@@ -0,0 +1,284 @@
+            )
+            if not errors:
+                return self.async_create_entry(
+                    title=f"{user_input[CONF_DEVICE]}", data=user_input
+                )
+        schema = _get_schema_common()
+        schema[vol.Required(CONF_RETAIN, default=True)] = bool
+        schema[vol.Required(CONF_TOPIC_IN_PREFIX)] = str
+        schema[vol.Required(CONF_TOPIC_OUT_PREFIX)] = str
+
+        schema = vol.Schema(schema)
+        return self.async_show_form(
+            step_id="gw_mqtt", data_schema=schema, errors=errors
+        )
+
+    def _normalize_persistence_file(self, path: str) -> str:
+        return os.path.realpath(os.path.normcase(self.hass.config.path(path)))
+
+    async def validate_common(
+        self,
+        gw_type: ConfGatewayType,
+        errors: Dict[str, str],
+        user_input: Optional[Dict[str, str]] = None,
+    ) -> Dict[str, str]:
+        """Validate parameters common to all gateway types."""
+        if user_input is not None:
+            errors.update(_validate_version(user_input.get(CONF_VERSION)))
+
+            if gw_type != CONF_GATEWAY_TYPE_MQTT:
+                if gw_type == CONF_GATEWAY_TYPE_TCP:
+                    verification_func = is_socket_address
+                else:
+                    verification_func = is_serial_port
+
+                try:
+                    await self.hass.async_add_executor_job(
+                        verification_func, user_input.get(CONF_DEVICE)
+                    )
+                except vol.Invalid:
+                    errors[CONF_DEVICE] = (
+                        "invalid_ip"
+                        if gw_type == CONF_GATEWAY_TYPE_TCP
+                        else "invalid_serial"
+                    )
+            if CONF_PERSISTENCE_FILE in user_input:
+                try:
+                    is_persistence_file(user_input[CONF_PERSISTENCE_FILE])
+                except vol.Invalid:
+                    errors[CONF_PERSISTENCE_FILE] = "invalid_persistence_file"
+                else:
+                    real_persistence_path = self._normalize_persistence_file(
+                        user_input[CONF_PERSISTENCE_FILE]
+                    )
+                    for other_entry in self.hass.config_entries.async_entries(DOMAIN):
+                        if CONF_PERSISTENCE_FILE not in other_entry.data:
+                            continue
+                        if real_persistence_path == self._normalize_persistence_file(
+                            other_entry.data[CONF_PERSISTENCE_FILE]
+                        ):
+                            errors[CONF_PERSISTENCE_FILE] = "duplicate_persistence_file"
+                            break
+
+            for other_entry in self.hass.config_entries.async_entries(DOMAIN):
+                if _is_same_device(gw_type, user_input, other_entry):
+                    errors["base"] = "already_configured"
```

Comment:
Please add this error key to the errors in `strings.json`.

If it's an import we should abort the flow. Can we add that?